pkg/kp/consulutil/session_controller.go

```diff
@@ -65,6 +68,11 @@ func SessionManager(
 				sessionLogger.NoFields().Info("session manager: released session")
 			}
 			output <- ""
+
+			// Wait a random amount of time between 0 and 10 seconds before attempting to
+			// re-establish a session. This protects against a thundering herd of sessions
+			// attempting to be re-establish
```

Comment:
`rand.Seed()` should only need to be called once per process to initialize the package's default entropy source. I suggest this call should go in `bin/p2-preparer/main.go` and be called with a hash of the current time and the node name or hostname.